app/controllers/projects_controller.rb

```diff
@@ -112,7 +112,11 @@ def require_project
   end
 
   def last_deploy_for(project)
-    @projects_last_deployed_at ||= Deploy.successful.last_deploys_for_projects
+    @projects_last_deployed_at ||= Deploy.successful.
+      last_deploys_for_projects.
+      includes(:project, job: :user).
```

Comment:
Haven't actually tried. I'm just accustomed to be explicit about that sort of thing to avoid N+1 queries.